### PR TITLE
Enable appveyor for CI

### PR DESCRIPTION
**Pull request by sgrif**
There's very little that had to change other than dicking around with
the configuration options. I had to release new pq-sys and
mysqlclient-sys versions that work better on Windows, but they were
patch bumps so they don't affect anything here.

For posterity and documenting how miserable today has been, I have left
the commit messages from each commit that I squashed to get this one in
tact below.

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

ಠ_ಠ

### CODE REVIEW

**Comment by sgrif** [comment]
I don't know what we need to do to make it show up as a status check, but it's definitely running. https://ci.appveyor.com/project/sgrif/diesel/build/1.0.95

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+install:
+  - appveyor DownloadFile https://win.rustup.rs/ -FileName rustup-init.exe
+  - rustup-init -yv --default-toolchain stable --default-host %target%
+  - set PATH=%PATH%;%USERPROFILE%\.cargo\bin
+  - rustc -vV
+  - cargo -vV
+  - appveyor DownloadFile https://sqlite.org/2017/sqlite-dll-win64-x64-3160200.zip -FileName sqlite3.zip
+  - 7z e sqlite3.zip -y
+  - set SQLITE3_LIB_DIR=%APPVEYOR_BUILD_FOLDER%
+build: false
```
Comment (killercup) [inline .appveyor.yml]: [Their reference](https://www.appveyor.com/docs/appveyor-yml/) says it means "disable automatic builds" but… what are "automatic builds" in this context? Is this about creating artifacts?

**Comment by killercup** [comment]
Oh and if this works, we might want to add some badges to the Cargo.toml files.

File: .appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
+version: 1.0.{build}
+
+# MySQL fails to link on 2015
+image: Visual Studio 2013
+
+branches:
+  only:
+    - master
+    - appveyor_nonsense
```
Comment (daschl) [inline .appveyor.yml]: is this still needed?

Comment (sgrif) [inline .appveyor.yml]: Yes, I keep a separate branch that I can just push to for when I need to debug CI things. On travis it's called `ಠ_ಠ` but Windows doesn't like unicode.

Comment (daschl) [inline .appveyor.yml]: 👍

**Pull request merged**